File: src/main/java/net/gtaun/shoebill/constant/ServerVarType.java
package net.gtaun.shoebill.constant;

/**
 * Created by marvin on 24.04.16.
 * Copyright (c) Marvin Haschker 2016.
 */
public enum ServerVarType {
    NONE (0),
    INTEGER (1),
    STRING (2),
    FLOAT (3);

    private int value;

    ServerVarType(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public static ServerVarType get(int id) {
        for (ServerVarType type : values()) {
            if (type.value == id)
                return type;
        }
        return null;
    }
}

File: src/main/java/net/gtaun/shoebill/constant/ServerVarType.kt
package net.gtaun.shoebill.constant

/**
 * This enum contains the available VarTypes for the ServerVar-System. This information was taken from the official
 * SA:MP Wikipedia page.
 *
 * @author Marvin Haschker
 */
enum class ServerVarType constructor(val value: Int) {
    NONE(0),
    INTEGER(1),
    STRING(2),
    FLOAT(3);

    companion object : Collectable<ServerVarType>, Findable<Int, ServerVarType> {
        private val VALUES = values().map { it.value to it }.toMap()

        /**
         * Gets all available [ServerVarType]s.
         */
        @JvmStatic
        override fun get(): Collection<ServerVarType> = VALUES.values

        /**
         * Gets a [ServerVarType] by its id.
         */
        @JvmStatic
        override operator fun get(value: Int): ServerVarType? = VALUES[value]

    }
}
